enshrine-esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -36,13 +39,139 @@ pub trait TransferTokensModule:
             sc_panic!("Operation is not registered");
         }
 
-        let minted_operation_tokens = self.mint_tokens(&operation.tokens);
+        let (remaining_tokens, is_wegld_fee_paid) = self.verify_operation_tokens_for_issue_fee(
+            &operation.data.op_sender,
+            operation.tokens.clone(),
+        );
+
+        if !is_wegld_fee_paid {
+            self.emit_transfer_failed_events(
+                &hash_of_hashes,
+                &OperationTuple {
+                    op_hash: operation_hash,
+                    operation,
+                },
+            );
+
+            sc_panic!("Not enough WEGLD to register all tokens");
+        }
+
+        let minted_operation_tokens = self.mint_tokens(&remaining_tokens);
         let operation_tuple = OperationTuple {
-            op_hash: operation_hash,
-            operation,
+            op_hash: operation_hash.clone(),
+            operation: operation.clone(),
+        };
+
+        self.distribute_payments(
+            hash_of_hashes.clone(),
+            operation_tuple,
+            minted_operation_tokens,
+        );
+    }
+
+    #[endpoint(registerTokens)]
+    #[payable("*")]
+    fn register_tokens(&self, tokens: MultiValueEncoded<TokenIdentifier>) {
+        let call_payment = self.call_value().single_esdt().clone();
+        let wegld_identifier = self.wegld_identifier().get();
+
+        require!(
+            call_payment.token_identifier == wegld_identifier,
+            "WEGLD is the only token accepted as register fee"
+        );
+
+        require!(
+            call_payment.amount == DEFAULT_ISSUE_COST * tokens.len() as u64,
+            "WEGLD fee amount is not met"
+        );
+
+        for token_id in tokens {
+            self.register_token(token_id);
+        }
+    }
+
+    #[endpoint(depositForNewRegistrations)]
+    #[payable("*")]
+    fn deposit_for_new_registrations(&self) {
```

Comment:
This endpoint is not doing anything. No WEGLD check, no registration, no nothing...

Reply:
My thoughts are that I either remove it altogether or I create another storage mapper that holds the current amount of WEGLD in the pool